packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1002,9 +1010,9 @@ export class TransactionController extends BaseController<
     }
     // 4. Pad estimated gas without exceeding the most recent block gasLimit. If the network is a
     // a custom network then return the eth_estimateGas value.
-    const gasBN = hexToBN(gasHex);
+    const gasBN = gasHex === undefined ? hexToBN('0x0') : hexToBN(gasHex);
```

Comment:
`hexToBN` now throws an error if given invalid input. It appears (from experimentation) that the previous behavior was to cast undefined implicitly to 0, so that's been preserved here.

Reply:
This is probably no longer needed due to fixing up the dependencies / `yarn.lock`, but this will bite us later, so I think it's worth keeping in.

Reply:
Removed here: [5589099](https://github.com/MetaMask/controllers/pull/831/commits/5589099bf3ac29330011b8a31b52e735f782331c)